src/app_service/service/settings/service.nim

```diff
@@ -58,6 +64,16 @@ QtObject:
         var statusUpdate = receivedData.currentStatus[0]
         self.events.emit(SIGNAL_CURRENT_USER_STATUS_UPDATED, CurrentUserStatusArgs(statusType: statusUpdate.statusType, text: statusUpdate.text))
 
+      if receivedData.settings.len > 0:
+        for settingsField in receivedData.settings:
+
+          if settingsField.name == KEY_CURRENCY:
+            self.settings.currency = settingsField.value
```

Comment:
This case I'll have to remove from this PR, as it tackles a different setting. But as I was building this, I ran into an issue which I've filed here #6092